packages/tslint-config/rules/enumSnakeCaseRule.ts

```diff
@@ -0,0 +1,43 @@
+import toSnakeCase = require('to-snake-case');
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = `Enum member names should be SCREAMING_SNAKE_CASE`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk);
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>): void {
+    // Recursively walk the AST starting with root node, `ctx.sourceFile`.
+    // Call the function `cb` (defined below) for each child.
+    return ts.forEachChild(ctx.sourceFile, cb);
+
+    function cb(node: ts.Node): void {
+        if (node.kind === ts.SyntaxKind.EnumMember && !isSnakeCase(node.getText(ctx.sourceFile))) {
+            return ctx.addFailureAtNode(node, Rule.FAILURE_STRING, getFix(node));
+        }
+        // Continue recursion into the AST by calling function `cb` for every child of the current node.
+        return ts.forEachChild(node, cb);
+    }
+
+    function getFix(node: ts.Node): Lint.Replacement {
+        const [key, value] = node
+            .getText(ctx.sourceFile)
+            .split('=')
```

Comment:
Tthis fails on some edge cases where `value` also contains one or more `=`.

For example:

```ts
export enum Operators {
     ASSIGN = '=',
     EQUALS = '==',
     TRIPLE_EQUALS = '===',
}
```

You can use this test to confirm:

```ts
    it('testing fixer with equals sign in value example', () => {
        const src = `enum Operators { Assign = '=', Equals = '==', TripleEquals = '===' }`;
        const expected = `enum Operators { ASSIGN = '=', EQUALS = '==', TRIPLE_EQUALS = '===' }`;
        const actual = getFixedResult(src, rule);
        const result = helper(src, rule);
        assert.equal(result.errorCount, 3); // tslint:disable-line:custom-no-magic-numbers
        assert.equal(actual, expected);
    });
```

The TypeScript parser has likely already sorted this out when constructing the AST. Is there a way for us to take advantage of the existing AST structure instead of trying to do a bit of manual parsing?